### PR TITLE
Allow developers to run pytest 

### PR DESCRIPTION
**Pull request by jayvdb**
This fixes two of the problems related to running pytest on developer machines (https://github.com/coala/coala/issues/4335)

Specifically:
- [x] https://github.com/coala/coala/issues/3511 - `py.test` fails due to version conflict of packages
- [x] https://github.com/coala/coala/issues/4336 - Set coverage environment variables automatically

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+coverage-config-reload-plugin~=0.1
 codecov~=2.0.5
 freezegun~=0.3.9
 pytest~=3.0
 pytest-cov~=2.2
 pytest-env~=0.6.0
+pytest-instafail~=0.3.0
 pytest-mock~=1.1
+pytest-reorder~=0.1.0
+pytest-reqs~=0.0.5
```
Comment (userzimmermann) [inline test-requirements.txt]: Update to `pytest-reqs~=0.0.6`. It makes config much easier :)

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from pytest_reqs import pytest_collection_modifyitems as _modifyitems
+
+
+def pytest_collection_modifyitems(config, session, items):
+    config.option.reqs = True
+    _modifyitems(config, session, items)
+    config.option.reqs = False
```
Comment (userzimmermann) [inline tests/conftest.py]: With the new `pytest-reqs-0.0.6` you can simplify the whole module to:

```python
from pytest_reqs import check_requirements


def pytest_collection_modifyitems(config, session, items):
    check_requirements(config, session, items)
```

File: test-requirements.txt (hunk 1 above)
Comment (jayvdb) [inline test-requirements.txt]: Ah wonderful, they fixed [it](https://github.com/di/pytest-reqs/issues/9) upstream.

Comment (userzimmermann) [inline test-requirements.txt]: @jayvdb Oh! That was done on your request... Even nicer :)

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4337/commits/4d4202598c5318a12393f66102af4a9830a8c058

**Comment by userzimmermann** [comment]
@rultor merge

**Comment by jayvdb** [comment]
reack c9e8d79 2138e82

**Comment by jayvdb** [comment]
@rultor merge

**Comment by thisch** [comment]
great! :+1:

**Comment by thisch** [comment]
Coverage should always be 100%, even if pytest is run locally, right?
The coverage on my machine is 99.96%, which is caused by the followed two files

```
coalib/output/ConsoleInteraction.py                                   258      0    104      1    99%   544->543
coalib/misc/CachingUtilities.py                                        75      1     20      1    98%   172, 169->172
```

**Comment by userzimmermann** [comment]
@thisch Are some tests somehow skipped in your local case?

**Comment by jayvdb** [comment]
There are a few holes in running pytest locally, which are being tracked on https://github.com/coala/coala/issues/4335

**Pull request merged**